client/service_create.go

```diff
@@ -58,6 +84,27 @@ func (cli *Client) ServiceCreate(ctx context.Context, service swarm.ServiceSpec,
 	return response, err
 }
 
+func getImageWithDigest(ctx context.Context, cli *Client, image, encodedAuth string) (string, []swarm.Platform, error) {
+	distributionInspect, err := cli.DistributionInspect(ctx, image, encodedAuth)
+	imageWithDigest := image
+	var platforms []swarm.Platform
+
+	if err == nil {
```

Comment:
How about returning early if there's an error? I don't really care much, but it might make the code more maintainable in the future.